src/webnn_native/Graph.h

```diff
@@ -62,10 +59,10 @@ namespace webnn_native {
         virtual MaybeError Finish();
 
       private:
-        ModelBase(ModelBuilderBase* modelBuilder, ObjectBase::ErrorTag tag);
-        virtual void CompileImpl(WebnnCompileCallback callback,
+        virtual void ComputeImpl(NamedInputsBase* inputs,
+                                 MLComputeCallback callback,
                                  void* userdata,
-                                 CompilationOptions const* options);
+                                 NamedOutputsBase* outputs) = 0;
```

Comment:
Probably we need to add a `CompileImpl` method and allow backends to implement. Please see my comments below.